src/io/http/src/main/scala/HTTPClients.scala

```diff
@@ -31,7 +30,9 @@ private[ml] trait HTTPClient extends BaseClient[HTTPRequestData, HTTPResponseDat
   }
 
   private[ml] def fromResponse(response: Response): ContextOut  = {
-    ContextOut(new HTTPResponseData(response.response), response.context)
+    val out = ContextOut(new HTTPResponseData(response.response), response.context)
+    response.response.close()
```

Comment:
This is an unusual convention so please check..
If a function is passed in (say) a stream, it is not usually responsible for closing it - it's typically the caller's responsibility. In fact, there are cases where closing an input stream inside a function might be the wrong thing to do (what if I want to do something with the stream later on)?

It might work in this particular case but this pattern is confusing and might be error-prone down the line.

Reply:
Will give this some more thinking, thanks!

Reply:
Refactored so this is better. This simplified alot so thanks for that!